engineapi/instance_manager.go

```diff
@@ -64,13 +66,21 @@ func CheckInstanceManagerCompatibilty(imMinVersion, imVersion int) error {
 }
 
 func CheckInstanceManagerProxyCompatibility(im *longhorn.InstanceManager) error {
-	if CurrentInstanceManagerProxyAPIVersion > im.Status.ProxyAPIVersion || CurrentInstanceManagerProxyAPIVersion < im.Status.ProxyAPIMinVersion {
+	if CurrentInstanceManagerProxyAPIVersion > im.Status.ProxyAPIVersion ||
+		CurrentInstanceManagerProxyAPIVersion < im.Status.ProxyAPIMinVersion {
 		return fmt.Errorf("current InstanceManager proxy version %v is not compatible with InstanceManagerProxyAPIVersion %v and InstanceManagerProxyAPIMinVersion %v",
 			CurrentInstanceManagerAPIVersion, im.Status.ProxyAPIVersion, im.Status.ProxyAPIMinVersion)
 	}
 	return nil
 }
 
+func CheckInstanceManagerProxySupport(im *longhorn.InstanceManager) error {
+	if UnsupportInstanceManagerProxyAPIVersion == im.Status.ProxyAPIVersion {
+		return fmt.Errorf("%v does not support proxy", im.Name)
+	}
+	return nil
+}
+
```

Comment:
nit: return a boolean value. The error message "... dost not suppport proxy" can move to caller side.

Reply:
I feel it should be consistent with [CheckInstanceManagerComptibility](https://github.com/longhorn/longhorn-manager/blob/master/engineapi/instance_manager.go#L58). WDYT?

Reply:
I see. I'm fine with it.